kore/src/Kore/Step/Step.hs

```diff
@@ -574,16 +595,27 @@ checkSubstitutionCoverage initial unified final
         ]
   where
     Predicated { term = axiom } = unified
+    unification = Predicated.toPredicate (Predicated.withoutTerm unified)
     leftAxiomVariables =
         Pattern.freeVariables leftAxiom
       where
         RulePattern { left = leftAxiom } = axiom
     Predicated { substitution } = final
-    substitutionVariables = Map.keysSet (Substitution.toMap substitution)
+    subst = Substitution.toMap substitution
+    substitutionVariables = Map.keysSet subst
     isCoveringSubstitution =
         Set.isSubsetOf leftAxiomVariables substitutionVariables
-    isSymbolic =
-        Foldable.any Target.isNonTarget substitutionVariables
+    isSymbolic = Foldable.any Target.isNonTarget substitutionVariables
+    isAcceptable = all isValidSymbolic (Map.toList subst)
+    isValidSymbolic (x, t)
+      | Target.isTarget x = True
+      | Valid.App_ symbolOrAlias _ <- t =
+        isConstructor symbolOrAlias || isSortInjection symbolOrAlias
```

Comment:
Maybe it would be worth checking that the entire pattern is made of constructor-like things (isConstructorLikePattern).